### ISSUE TITLE
Lab 2 - Question about the loss function

### ISSUE THREAD

**Issue by thangnh0608** [eje24/iap-diffusion-labs#21]
In the course code, the `get_train_loss` function uses this loss formulation, which is exactly the same loss function in the lecture notes:

```python
mse = torch.sum(torch.square(s_theta - s_ref), dim=-1)
loss = torch.mean(mse)
```

I’m wondering if my MSE loss function is written as this:

```python
mse = torch.square(s_theta - s_ref)
loss = torch.mean(mse)
```
I tried it and it works fine. My understanding is that this computes loss per dimension rather than per sample. Just to confirm, mathematically, is that OK to use my implementation of the MSE loss? 

Thank you!

**Comment by eje24** [comment]
Hi @thangnh0608 , that's right. They're equivalent up to a constant (in this case, the data dimensionality), so you'd want to ensure that you handle the learning rate accordingly.
